workflows/rnasplice.nf

```diff
@@ -61,23 +123,494 @@ def multiqc_report = []
+                    multiple: fastq.size() > 1
+                        return [ meta, fastq.flatten() ]
+            }
+            .set { ch_fastq }
+            ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+            break;
+        case 'genome_bam':
+            INPUT_CHECK (
+                ch_input
+            )
+            .out
+            .set { ch_genome_bam }
+            ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+            break;
+        case 'transcriptome_bam':
+            INPUT_CHECK (
+                ch_input
+            )
+            .out
+            .set { ch_transcriptome_bam }
+            ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+            break;
+        case 'salmon_results':
+            INPUT_CHECK (
+                ch_input
+            )
+            .out
+            .set { ch_salmon_results }
+            ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+            break;
+    }
+
+    // Create samplesheet channel (after input check)
+    ch_samplesheet = Channel.fromPath(params.input)
+
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
+    // SUBWORKFLOW: Read in contrastsheet, validate and stage input files
+    //
+    CONTRASTS_CHECK (
+        ch_contrasts
     )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+    ch_versions = ch_versions.mix(CONTRASTS_CHECK.out.versions)
+
+    // Create contrastsheet channel (after contrasts check)
+    ch_contrastsheet = Channel.fromPath(params.contrasts)
+
+
+    // Check rMATS parameters specified correctly
+    if (params.rmats && params.source == 'fastq') {
+            WorkflowRnasplice.rmatsReadError(INPUT_CHECK.out.reads, log)
+            WorkflowRnasplice.rmatsStrandednessError(INPUT_CHECK.out.reads, log)
+    }
+
+    //
+    // MODULE: Concatenate FastQ files from same sample if required
+    //
+    if (params.source == 'fastq') {
+        CAT_FASTQ (
+            ch_fastq.multiple
+        )
+        .reads
+        .mix(ch_fastq.single)
+        .set { ch_cat_fastq }
+        ch_versions = ch_versions.mix(CAT_FASTQ.out.versions)
+    }
+
+    //
+    // SUBWORKFLOW: Read QC and trim adapters with TrimGalore!
+    //
+    if (params.source == 'fastq') {
+        FASTQC_TRIMGALORE (
+            ch_cat_fastq,
+            params.skip_fastqc,
+            params.skip_trimming
+        )
+        ch_trim_reads = FASTQC_TRIMGALORE.out.reads
+        ch_versions = ch_versions.mix(FASTQC_TRIMGALORE.out.versions)
+    }
+
+    //
+    // SUBWORKFLOW: Alignment with STAR
+    //
+
+    ch_genome_bam                 = Channel.empty()
+    ch_genome_bam_index           = Channel.empty()
+    ch_samtools_stats             = Channel.empty()
+    ch_samtools_flagstat          = Channel.empty()
+    ch_samtools_idxstats          = Channel.empty()
+    ch_star_multiqc               = Channel.empty()
+
+    if (params.source == 'genome_bam') {
+
+        BAM_SORT_SAMTOOLS ( ch_genome_bam )
```

Comment:
If you use the nf-core subworkflows this can be done inside the subworkflow

Reply:
Agreed, I like how nf-core/cutandrun is organised so will look to refactor similarity in next release.